### PR TITLE
project monitor tile

### PR DESCRIPTION
**Pull request by aviatesk**
- [x] basic implementation
- [x] compat stuff
- [x] initialization stuff
- [x] `Pkg` mode hack
- [x] test

NOTE: require v1.4 for getting this feature to work

### CODE REVIEW

**Comment by aviatesk** [comment]
okay I've done most basic stuff, and only the thing remaining is writing test.
I would welcome reviews on this.

File: src/Atom.jl [hunk 1]
```diff
@@ -12,9 +12,28 @@ function __init__()
...
+    i === nothing && return # if no Pkg mode somehow
+    pkg_mode = Base.active_repl.interface.modes[i]
+    pkg_mode.on_done = let original = pkg_mode.on_done
+      (s, buf, ok) -> begin
+        original(s, buf, ok)
+        update_project()
+      end
+    end
+  end
+  push!(Base.repl_hooks, f)
```
[the remark this answers is not in the document]
Comment (aviatesk) [reply src/Atom.jl:36]: unfortunately not, this overwriting only works after Pkg.jl's repl mode is created into `Base.active_repl.interface`, but since Pkg.jl is initialized before Atom.jl (and register its `atreplinit` hook before us), this function would be called before Pkg.jl's `atreplinit` hook if we just use `atreplinit`.

This is because `atreplinit` uses [`pushfirst`](https://github.com/JuliaLang/julia/blob/097f0b0c8612b4b172e403de2b852d962fb7bb6d/base/client.jl#L342) for registering a hook function.

File: src/environments.jl [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+@static if VERSION < v"1.1"
+
+update_project() = @msg updateProject(false)
+handle(()->false, "allProjects")
+
+else
+
+update_project() = @msg updateProject(project_info())
+
+atinit(update_project)
```
[the remark this answers is not in the document]
Comment (aviatesk) [reply src/environments.jl:24]: at the point of `__init__`, we haven't established a connection, so `update_project()` won't work at that time.

[the remark this answers is not in the document]
Comment (pfitzseb) [reply src/environments.jl:24]: I meant the call to `atinit`, not `update_project`.
Usually we're doing this kind of init stuff (this one and the one above) on the Atom side, which might be an option too.

[the remark this answers is not in the document]
Comment (aviatesk) [reply src/environments.jl:24]: ah I see, yeah this is better to be moved into `__init__`

[the remark this answers is not in the document]
Comment (aviatesk) [reply src/environments.jl]: I remove `atinit` stuff at all and moved `update_project` into `connected` handler.

> Usually we're doing this kind of init stuff (this one and the one above) on the Atom side, which might be an option too.

yeah, doing this initialization in Atom would be better in terms of configuring this feature. 
sounds like I should modify the code a bit

[the remark this answers is not in the document]
Comment (aviatesk) [reply src/environments.jl]: > sounds like I should modify the code a bit

hmm but for this case, what sends a message should be Julia (since we don't know when to update an environment information in Atom side), and we need to scatter around `update_project` as PR does anyway.

So for configuring this feature, what we can do is to turn on/off [Atom side handler](https://github.com/JunoLab/atom-julia-client/blob/42264dec40c95c128fbd2d885d37a83b83738987/lib/runtime/environments.js#L33-L40) while we need to receive messages from Julia anyway.

So in conclusion I think this PR is okay as is.

File: src/environments.jl [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+else
+
+update_project() = @msg updateProject(project_info())
+
+# IDEA: Pkg.project will be useful for this, but it's only available as of v1.4
+function project_info()
+  m = match(r"Status \`(?<path>.+)\`$"m, project_status())
+  m === nothing && return false
+  return project_info(m[:path])
+end
```
Comment (aviatesk) [inline src/environments.jl:29]: Do you think is it okay to drop support for <v1.4 ? Then we can just use `Pkg.project` for `project_info` implementation and don't need to use the maybe-non-robust regex match.

Comment (aviatesk) [reply src/environments.jl]: in `project_info` I also use `Pkg.REPLMode.projname`, which is so internal, and I'm not sure from which version those all the Pkg.jl functions are available.
I would like to drop support for <v1.3.
